engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodImportResolver.java

```diff
@@ -0,0 +1,139 @@
+import org.enso.interpreter.runtime.Module;
+import org.enso.interpreter.runtime.callable.UnresolvedSymbol;
+import org.enso.interpreter.runtime.data.EnsoObject;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.scope.TopLevelScope;
+
+final class InvokeMethodImportResolver
+    extends ImportResolverAlgorithm<EnsoObject, Module, UnresolvedSymbol, Object, Type, Module> {
+
+  private final Module module;
+  private final TopLevelScope topScope;
+
+  private InvokeMethodImportResolver(Module module, TopLevelScope topScope) {
+    this.module = module;
+    this.topScope = topScope;
+  }
+
+  @Override
+  protected String nameForImport(UnresolvedSymbol name) {
+    var fqn = module.getName().pathAsJava();
+    String moduleName;
+    if (fqn.size() == 2 && "Main".equals(module.getName().item())) {
+      moduleName = module.getPackage().libraryName().toString();
+    } else {
+      moduleName = module.getName().toString();
+    }
+    var subModuleName = moduleName + "." + name.getName();
+    return subModuleName;
+  }
+
+  @Override
+  protected List<String> partsForImport(UnresolvedSymbol imp) {
+    return module.getName().createChild(imp.getName()).createChild("any").pathAsJava();
+  }
+
+  @Override
+  protected String nameForExport(Object ex) {
+    throw new AssertionError("not used: " + ex);
+  }
+
+  @Override
+  protected String nameForType(Type e) {
+    return e.getName();
+  }
+
+  @Override
+  protected List<Object> exportsFor(Module module, String impName) {
+    return Collections.emptyList();
+  }
+
+  @Override
+  protected boolean isAll(Object ex) {
+    return false;
+  }
+
+  @Override
+  protected List<String> onlyNames(Object ex) {
+    return null;
+  }
+
+  @Override
+  protected List<String> hiddenNames(Object ex) {
+    return null;
+  }
+
+  @Override
+  protected List<Type> definedEntities(UnresolvedSymbol name) {
+    var associatedType = module.getScope().getType(name.getName());
+    var allRelativeTypes = module.getScope().getTypes().values();
+    return Stream.concat(associatedType.stream(), allRelativeTypes.stream()).toList();
+  }
+
+  @Override
+  protected Module loadLibraryModule(LibraryName libraryName, String moduleName)
+      throws IOException {
+    var optionModule = topScope.getModule(moduleName);
+    return optionModule.isPresent() ? optionModule.get() : null;
+  }
+
+  @Override
+  protected EnsoObject createResolvedImport(UnresolvedSymbol imp, List<Object> exp, Module m) {
+    return m.getScope().getAssociatedType();
+  }
+
+  @Override
+  protected EnsoObject createResolvedType(UnresolvedSymbol imp, List<Object> exp, Type typ) {
+    return typ;
+  }
+
+  @Override
+  protected EnsoObject createErrorPackageCoundNotBeLoaded(
+      UnresolvedSymbol imp, String impName, String loadingError) {
+    return null;
+  }
+
+  @Override
+  protected EnsoObject createErrorModuleDoesNotExist(UnresolvedSymbol imp, String impName) {
+    return null;
+  }
```

Comment:
why are those returning `null`?

Reply:
`null` means _not found_ and then [methodNotFound](https://github.com/enso-org/enso/pull/9539/files#diff-6bc7c2f8a27bdb61f738a75b6accb867d8b5336e3f7d1ea3c7128b26f36e2fa6R216) exception will be thrown.